backends-common/postgres/src/main/java/org/apache/james/backends/postgres/utils/PoolPostgresExecutor.java

```diff
@@ -0,0 +1,152 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.backends.postgres.utils;
+
+import java.util.Optional;
+import java.util.function.Function;
+
+import javax.inject.Inject;
+
+import org.apache.james.backends.postgres.PostgresConfiguration;
+import org.apache.james.lifecycle.api.Disposable;
+import org.jooq.DSLContext;
+import org.jooq.Record;
+import org.jooq.Record1;
+import org.jooq.impl.DSL;
+import org.reactivestreams.Publisher;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.r2dbc.pool.ConnectionPool;
+import io.r2dbc.pool.ConnectionPoolConfiguration;
+import io.r2dbc.spi.Connection;
+import io.r2dbc.spi.ConnectionFactory;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+public class PoolPostgresExecutor implements PostgresExecutor, Disposable {
```

Comment:
My feeling is this `PoolPostgresExecutor` introduction can be avoided, and we just use a single `PostgresExecutor`.

We already have this code:
```java
    @Provides
    @Singleton
    JamesPostgresConnectionFactory provideJamesPostgresConnectionFactory(PostgresConfiguration postgresConfiguration, ConnectionFactory connectionFactory) {
        if (postgresConfiguration.rowLevelSecurityEnabled()) {
            LOGGER.info("PostgreSQL row level security enabled");
            LOGGER.info("Implementation for PostgreSQL connection factory: {}", DomainImplPostgresConnectionFactory.class.getName());
            return new DomainImplPostgresConnectionFactory(connectionFactory);
        }
        LOGGER.info("Implementation for PostgreSQL connection factory: {}", SinglePostgresConnectionFactory.class.getName());
        return new SinglePostgresConnectionFactory(Mono.from(connectionFactory.create()).block());
    }
```

Maybe we can modify `SinglePostgresConnectionFactory` to be backed by the pool connection (and rename it), then avoid the `PoolPostgresExecutor` introduction?

Reply:
how do we can "PoolConnection::close" (that releases the connection to the pool after query)?
I don't think we like invoking `.close` method everywhere

Reply:
> Maybe we can modify SinglePostgresConnectionFactory to be backed by the pool connection (and rename it)

+1 that's what I had in mind - and tried to explain: let's see if it makes a difference ;-)

My advice: untungle the two topics:
 - Merge the blobStore part without the pool, disabling tests as needed
 - Create a ticket to design a proper PGExecutor, don't rush it, propose several designs, try to implement them, and bench them, but later...

Reply:
The smaller split of only PostgresBlobStorage
https://github.com/apache/james-project/pull/1895

Reply:
We can discuss about designs in https://github.com/linagora/james-project/issues/5020